clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10266,6 +10266,11 @@ def err_block_on_vm : Error<
 def err_sizeless_nonlocal : Error<
   "non-local variable with sizeless type %0">;
 
+def err_vec_builtin_non_vector_all : Error<
+ "all arguments to %0 must be vectors">;
+def err_vec_builtin_incompatible_vector_all : Error<
+  "all arguments to %0 must have vectors of the same type">;
```

Comment:
It wouldn't make sense to change it as part of this commit, but with this in place "err_vec_builtin_non_vector" has a bit of an ambiguous name now. Might be worth following up at some point

Reply:
I did it this way because the amount of test cases I would need to update to not be tied to two args was very large.

Reply:
I disagree here. I think we should collapse these down. You can use a `%select{...}` block in the error format string to support the old message and the new message from the same diagnostic entry.

These diagnostic strings end up becoming some pretty large constant data tables in the final clang executable so we've generally tried to reuse errors where it makes sense.

Reply:
https://github.com/llvm/llvm-project/pull/83609